packages/nimble-components/src/dialog/tests/dialog.stories.ts

```diff
@@ -23,7 +23,7 @@ interface DialogArgs {
     openAndHandleResult: (
         dialogRef: Dialog<string>,
         textFieldRef: TextField
-    ) => void;
+    ) => Promise<void> | undefined;
```

Comment:
For my own edification, what's the purpose of declaring that this can return `undefined`?

Reply:
There is one case in the tests, [on line 113](https://github.com/ni/nimble/blob/cff34b118751596ac7f85590eb287e3d3ca27f79/packages/nimble-components/src/dialog/tests/dialog.stories.ts#L113), where it returns undefined.